### DISCUSSION TITLE
REST API Spec Markdown Emitter

### DISCUSSION THREAD

**Discussion by wangyuantao** [microsoft/typespec#2793] [General]
I'm wondering if there is any existing Markdown Emitter?

The output .md can be from a build-in template, like "Microsoft Learn" [style](https://learn.microsoft.com/en-us/rest/api/cognitiveservices/accountmanagement/operation-groups).

If not, how can I contribute one? It seems TypeSpec provide the [framework](https://microsoft.github.io/typespec/extending-typespec/emitters)?

I'm not considering "TypeSpec → emit OpenAPI (2 or 3) → Markdown" path, because the intermediate OpenAPI loss TypeSpec fidality.

**Comment by timotheeguerin** [comment]
Hi @wangyuantao would you mean a markdown emitter in the sense it would document the API? We do not have that for APIs today(The microsoft docs are built from swagger still). However our own reference docs are generated using this package which is not public yet but you if are intrested you can base yourself of that one. https://github.com/microsoft/typespec/tree/main/packages/tspd/src/ref-doc
